src/main/java/de/adesso/blogpostchecker/FileAnalyzer.java

```diff
@@ -114,9 +114,16 @@ private void extractMetadataFromFiles(RevCommit latestCommit, DiffEntry blogPost
     }
 
     private Ref getBranchByName() throws GitAPIException {
-        return localGitInstance.branchList().setListMode(ListBranchCommand.ListMode.REMOTE).call()
+        Ref branch = localGitInstance.branchList().setListMode(ListBranchCommand.ListMode.REMOTE).call()
                 .stream().filter(ref -> extractBranchName(ref).equals(configService.getREPOSITORY_BRANCH_NAME()))
                 .findFirst().orElse(null);
+
+        if (branch == null) {
+            branch = localGitInstance.branchList().setListMode(ListBranchCommand.ListMode.ALL).call()
+                    .stream().filter(ref -> extractBranchName(ref).equals(configService.getREPOSITORY_BRANCH_NAME()))
+                    .findFirst().orElse(null);
```

Comment:
Das sieht schon hart effizient aus, aber ich kann leider nur sehr sehr schwer nachvollziehen was dort passiert.
Eines Tages würde das niemand mehr anfassen wollen, weil es "funktioniert".

Kannst du versuchen die (gesamte) Methoden zu vereinfachen oder zu trennen? Wenn das ein bisschen mehr Code erzeugt wäre es das wert. Zudem scheint sich ein breiterer Teil zu wiederholen. Vlt. können wir diese Stelle als Methode auskappseln und abhängig vom `ListBranchCommand.ListMode` ausführen. In dem Fall wäre es vlt. in Ordnung die extrahierte Methode mit einem aussagekräftigen Namen und Kommentar so zu belaßen.